server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1359,6 +1398,7 @@ public void onFailure(String source, Exception e) {
 
                     @Override
                     public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
+                        cacheRepositoryData(filteredRepositoryData.withGenId(newGen));
```

Comment:
This is a little dirty and we talked about it in other PRs .. the generation here is `-1` due to the weird way the repository data is loaded initially and we have to eventually set it. I'll clean that up in a follow up, for now just setting it here where it matters (it doesn't matter in the above code when serializing `filteredRepositoryData` to the repo) seemed the driest.

Reply:
Maybe add a TODO then?

Reply:
Not sure what to even put in it though. In the end, this is simply the way `RepositoryData` works for now. We have the same kind of code for the cluster state version as well I guess. I don't have a good idea for a better abstract yet :(